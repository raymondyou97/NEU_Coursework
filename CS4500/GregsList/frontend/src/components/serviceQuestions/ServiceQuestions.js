import React from 'react';
import '../../styles/serviceQA.css';

class ServiceQuestions extends React.Component {
  componentDidMount() {
    this.props.fetchServiceQuestions();
  }

  handleQuestionChange = e => {
    this.props.handleQuestionChange(e);
  };

  handleTypeChange = e => {
    this.props.handleTypeChange(e);
  };

  handleChoicesChange = e => {
    this.props.handleChoicesChange(e);
  };

  deleteServiceQuestion = id => {
    this.props.deleteServiceQuestion(id);
  };

  render() {
    let { selectedServiceQuestion, serviceQuestionIDChoices } = this.props;

    return (
      <div className="service-question-table">
        <h3>Service Questions</h3>
        <div>
          Note: Types are (TrueFalse, MultipleChoice, Range) and choices are
          Comma-separated
        </div>
        <table className="service-q-a-table">
          <thead className="service-q-a-table-header">
            <tr className="service-q-a-table-row">
              <th>ID</th>
              <th>Question</th>
              <th>Type</th>
              <th>Choices</th>
              <th>Commands</th>
            </tr>
          </thead>
          <tbody className="service-q-a-tableBody">
            <tr className="service-q-a-input-row">
              <td>{serviceQuestionIDChoices}</td>
              <td>
                <input
                  type="text"
                  placeholder="Question"
                  value={selectedServiceQuestion.newQuestion}
                  className="service-question-input-question"
                  onChange={this.handleQuestionChange}
                />
              </td>
              <td>
                <input
                  type="text"
                  placeholder="Type"
                  value={selectedServiceQuestion.newType}
                  className="service-question-input-type"
                  onChange={this.handleTypeChange}
                />
              </td>
              <td>
                <input
                  type="text"
                  placeholder="Choices"
                  value={selectedServiceQuestion.newChoices}
                  className="service-question-input-choices"
                  onChange={this.handleChoicesChange}
                />
              </td>
              <td>
                <button
                  className="service-q-a-add-button"
                  onClick={() => this.props.createServiceQuestion()}
                >
                  Add
                </button>
                <button
                  className="service-q-a-update-button"
                  onClick={() => this.props.updateServiceQuestion()}
                >
                  Update
                </button>
              </td>
            </tr>
            {this.props.serviceQuestions.map(serviceQuestion => (
              <tr key={serviceQuestion.id} className="service-q-a-table-row">
                <td>{serviceQuestion.id}</td>
                <td>{serviceQuestion.question}</td>
                <td>{serviceQuestion.type}</td>
                <td>{serviceQuestion.choices}</td>
                <td>
                  <button
                    className="service-q-a-show-button"
                    onClick={() =>
                      this.props.showServiceQuestionDetails(serviceQuestion.id)
                    }
                  >
                    Show
                  </button>
                  <button
                    className="service-q-a-delete-button"
                    onClick={() =>
                      this.props.deleteServiceQuestion(serviceQuestion.id)
                    }
                  >
                    Delete
                  </button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
        <div className="pagination-row">
          <select onChange={this.props.handleCountChange}>
            <option value={10}>10</option>
            <option value={25}>25</option>
            <option value={50}>50</option>
            <option value={100}>100</option>
            <option value={'All'}>All</option>
          </select>
          <button className="pagination-button" onClick={this.props.pageLeft}>
            Prev
          </button>
          <button className="pagination-button" onClick={this.props.pageRight}>
            Next
          </button>
        </div>
      </div>
    );
  }
}

export default ServiceQuestions;
