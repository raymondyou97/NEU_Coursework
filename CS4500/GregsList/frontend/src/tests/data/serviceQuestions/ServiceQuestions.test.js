'use es6';

import React from 'react';
import ServiceQuestions from '../../../components/serviceQuestions/ServiceQuestions';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SERVICE_QUESTION_LIST } from './ServiceQuestionData';

configure({ adapter: new Adapter() });

test('Snapshot renders', () => {
  let selectedServiceQuestion = {
    newQuestion: '',
    newType: '',
    newChoices: '',
  };

  const component = renderer.create(
    <ServiceQuestions
      serviceQuestions={SERVICE_QUESTION_LIST}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot renders service questions component updates when there is data entered into the input boxes', () => {
  let selectedServiceQuestion = {
    newQuestion: 'what is love?',
    newType: 'what is life?',
    newChoices: 'what is death?',
  };

  const component = renderer.create(
    <ServiceQuestions
      serviceQuestions={SERVICE_QUESTION_LIST}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Initial DOM contains correct number of rows of service questions including input row', () => {
  let selectedServiceQuestion = {
    newQuestion: '',
    newType: '',
    newChoices: '',
  };

  let component = shallow(
    <ServiceQuestions
      serviceQuestions={[]}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  expect(
    component.find('.service-question-table .service-q-a-table-row').length
  ).toEqual(1);

  component = shallow(
    <ServiceQuestions
      serviceQuestions={SERVICE_QUESTION_LIST}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  expect(
    component.find('.service-question-table .service-q-a-table-row').length
  ).toEqual(4);
});

test('Service question component input value changes if input is entered', () => {
  let selectedServiceQuestion = {
    newQuestion: '',
    newType: '',
    newChoices: '',
  };

  let component = shallow(
    <ServiceQuestions
      serviceQuestions={SERVICE_QUESTION_LIST}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  expect(
    component.find('.service-question-input-question').props().value
  ).toEqual('');
  expect(component.find('.service-question-input-type').props().value).toEqual(
    ''
  );
  expect(
    component.find('.service-question-input-choices').props().value
  ).toEqual('');

  selectedServiceQuestion = {
    newQuestion: 'what is love?',
    newType: 'what is life?',
    newChoices: 'what is death?',
  };

  component = shallow(
    <ServiceQuestions
      serviceQuestions={SERVICE_QUESTION_LIST}
      selectedServiceQuestion={selectedServiceQuestion}
      serviceQuestionIDChoices={[]}
      fetchServiceQuestions={() => {}}
      handleQuestionChange={() => {}}
      handleTypeChange={() => {}}
      handleChoicesChange={() => {}}
      handleCountChange={() => {}}
      createServiceQuestion={() => {}}
      showServiceQuestionDetails={() => {}}
      deleteServiceQuestion={() => {}}
      updateServiceQuestion={() => {}}
      getServiceQuestionIDs={() => {}}
      pageLeft={() => {}}
      pageRight={() => {}}
    />
  );

  expect(
    component.find('.service-question-input-question').props().value
  ).toEqual('what is love?');
  expect(component.find('.service-question-input-type').props().value).toEqual(
    'what is life?'
  );
  expect(
    component.find('.service-question-input-choices').props().value
  ).toEqual('what is death?');
});
