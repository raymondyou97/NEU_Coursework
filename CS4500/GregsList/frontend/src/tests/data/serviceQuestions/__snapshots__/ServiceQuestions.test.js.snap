// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div
  className="service-question-table"
>
  <h3>
    Service Questions
  </h3>
  <div>
    Note: Types are (TrueFalse, MultipleChoice, Range) and choices are Comma-separated
  </div>
  <table
    className="service-q-a-table"
  >
    <thead
      className="service-q-a-table-header"
    >
      <tr
        className="service-q-a-table-row"
      >
        <th>
          ID
        </th>
        <th>
          Question
        </th>
        <th>
          Type
        </th>
        <th>
          Choices
        </th>
        <th>
          Commands
        </th>
      </tr>
    </thead>
    <tbody
      className="service-q-a-tableBody"
    >
      <tr
        className="service-q-a-input-row"
      >
        <td />
        <td>
          <input
            className="service-question-input-question"
            onChange={[Function]}
            placeholder="Question"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="service-question-input-type"
            onChange={[Function]}
            placeholder="Type"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="service-question-input-choices"
            onChange={[Function]}
            placeholder="Choices"
            type="text"
            value=""
          />
        </td>
        <td>
          <button
            className="service-q-a-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-q-a-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          1
        </td>
        <td>
          question1
        </td>
        <td>
          type1
        </td>
        <td>
          choice1
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          2
        </td>
        <td>
          question2
        </td>
        <td>
          type2
        </td>
        <td>
          choice2
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          3
        </td>
        <td>
          question3
        </td>
        <td>
          type3
        </td>
        <td>
          choice3
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="pagination-row"
  >
    <select
      onChange={[Function]}
    >
      <option
        value={10}
      >
        10
      </option>
      <option
        value={25}
      >
        25
      </option>
      <option
        value={50}
      >
        50
      </option>
      <option
        value={100}
      >
        100
      </option>
      <option
        value="All"
      >
        All
      </option>
    </select>
    <button
      className="pagination-button"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="pagination-button"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Snapshot renders service questions component updates when there is data entered into the input boxes 1`] = `
<div
  className="service-question-table"
>
  <h3>
    Service Questions
  </h3>
  <div>
    Note: Types are (TrueFalse, MultipleChoice, Range) and choices are Comma-separated
  </div>
  <table
    className="service-q-a-table"
  >
    <thead
      className="service-q-a-table-header"
    >
      <tr
        className="service-q-a-table-row"
      >
        <th>
          ID
        </th>
        <th>
          Question
        </th>
        <th>
          Type
        </th>
        <th>
          Choices
        </th>
        <th>
          Commands
        </th>
      </tr>
    </thead>
    <tbody
      className="service-q-a-tableBody"
    >
      <tr
        className="service-q-a-input-row"
      >
        <td />
        <td>
          <input
            className="service-question-input-question"
            onChange={[Function]}
            placeholder="Question"
            type="text"
            value="what is love?"
          />
        </td>
        <td>
          <input
            className="service-question-input-type"
            onChange={[Function]}
            placeholder="Type"
            type="text"
            value="what is life?"
          />
        </td>
        <td>
          <input
            className="service-question-input-choices"
            onChange={[Function]}
            placeholder="Choices"
            type="text"
            value="what is death?"
          />
        </td>
        <td>
          <button
            className="service-q-a-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-q-a-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          1
        </td>
        <td>
          question1
        </td>
        <td>
          type1
        </td>
        <td>
          choice1
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          2
        </td>
        <td>
          question2
        </td>
        <td>
          type2
        </td>
        <td>
          choice2
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          3
        </td>
        <td>
          question3
        </td>
        <td>
          type3
        </td>
        <td>
          choice3
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="pagination-row"
  >
    <select
      onChange={[Function]}
    >
      <option
        value={10}
      >
        10
      </option>
      <option
        value={25}
      >
        25
      </option>
      <option
        value={50}
      >
        50
      </option>
      <option
        value={100}
      >
        100
      </option>
      <option
        value="All"
      >
        All
      </option>
    </select>
    <button
      className="pagination-button"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="pagination-button"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
